src/log/log.c

```diff
@@ -1854,6 +1916,31 @@ err:	__wt_free(session, buf);
 	return (ret);
 }
 
+/*
+ * __log_check_partial_write --
+ *	Determine if the log record is not a partial write. A partial write
+ *	is initially detected by a mismatched checksum. A partial write
+ *	will have zeroes for some portion at the end.  If there are no zeroes,
+ *	the write cannot be partial, and the checksum error must be the
+ *	result of corruption.
+ */
+static void
+__log_check_partial_write(WT_SESSION_IMPL *session, WT_ITEM *buf,
+    uint32_t reclen, bool *corrupt)
+{
+	uint8_t *rec;
+
+	WT_UNUSED(session);
+
+	/*
+	 * We only check the final byte. An actual partial write would likely
+	 * end on an aligned memory boundary, but I don't think there are any
```

Comment:
Since this is the real record length, I think you're correct about the alignment. But the record overall, in the log file, is rounded up to the `log->allocsize` boundary. Not sure if that is helpful or not.

Reply:
Added more flesh to the comment to explain.